### PR TITLE
Change targetType from 'sourceLibrary' to the implicit 'library'.

### PR DESCRIPTION
**Pull request by p0nce**
I'm not confident about using `sourceLibrary` in DUB.
For one, IDE projects generated by DUB for Sublime Text or VisualD don't show the imageformats source files.
Secondly, the advantage of compiling all-at-once are still there with `--combined`.

### CODE REVIEW

**Comment by sigod** [comment]
> Secondly, the advantage of compiling all-at-once are still there with `--combined`.

Can you provide some links about `--combined`? For some reason I can't find anything. Is it DUB's switch?

**Comment by p0nce** [comment]
Yes, it is documented on `dub run -h`, `dub build -h` or  `dub generate -h`.

`dub generate --combined visuald` will generate all projects in a single .vcxproj, and similarly for Sublime.

`--combined` build tends to have more problems when the different packages have flags requirements.
But `sourceLibrary` would be affected as well. My angst with `sourceLibrary`is that I don't see many people use it, and I can almost feel the future problems coming our way :)

**Comment by sigod** [comment]
> Yes, it is documented on `dub run -h`, `dub build -h` or `dub generate -h`.
> 
> `dub generate --combined visuald` will generate all projects in a single .vcxproj, and similarly for Sublime.

Never knew about it. Thanks.

> and I can almost feel the future problems coming our way :)

You mean significant compilation time?

**Comment by p0nce** [comment]
No no, build problems. (Compilation time should be better for `sourceLibrary` and `--combined`)

For example, let's say I have a compiler codegen bug in LDC 17.0.x. By compiling in separate libraries I can use a DUB variable to disable optimization for this package, and done. But such a work-around doesn't work with `--combined` and `sourceLibrary`. They break locality of compile bugs because now any user package is affected. Example of such: https://github.com/p0nce/dplug/issues/35

Basically I'm advocating for more separate compilation, because using `sourceLibrary` can reveal annoying because it limits choice for building. And it is very much a problem in my experience, `--combined`-type builds are easier to break.

**Comment by p0nce** [comment]
Thanks.

**Comment by sigod** [comment]
Interesting. I need to dig into this. Thanks.
